### ISSUE TITLE
[functional/immutable-data] Doesn't take into consideration ignoreAccessorPattern with hash symbol

### ISSUE THREAD

**Issue by Clarchik** [eslint-functional/eslint-plugin-functional#911]
## Bug Report

Seems like rule does not work as intended.

### Expected behavior
Rule below should not produce error for variables which follows # symbol (TS private modifier)
```
'functional/immutable-data': [
      'error',
      {
        ignoreAccessorPattern: ['**._*', '**.#*']
      }
    ],
```

### Actual behavior

Works for 

`this._anyName = value`

But produces error for

`this.#anyName = value`

`Modifying an existing object/array is not allowed.eslint[functional/immutable-data]`

### Steps to reproduce

Latest version (7.2.0) with the rule snippet above

**Comment by RebeccaStevens** [comment]
See https://github.com/eslint-functional/eslint-plugin-functional/issues/910#issuecomment-2571522917
